libs/sdk-common/src/test_utils/mock_server_wasm.rs

```diff
@@ -0,0 +1,101 @@
+use js_sys::*;
+use reqwest::Client;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::sync::Mutex;
+use wasm_bindgen::{prelude::*, JsCast};
+use wasm_bindgen_futures::JsFuture;
+use web_sys::*;
+
+#[wasm_bindgen(module = "/mock_service_worker.js")]
+extern "C" {}
+
+#[derive(Debug, Serialize)]
+struct MockConfig {
+    pub kind: String,
+    pub method: String,
+    pub path: String,
+    pub status_code: u16,
+    pub content_type: Option<String>,
+    pub body: Vec<u8>,
+}
+
+pub(crate) struct MockServer {
+    client: Mutex<reqwest::Client>,
+    url_map: Mutex<HashMap<String, String>>,
+}
+
+#[allow(dead_code)]
+pub(crate) struct ServerOpts {
+    pub host: &'static str,
+    pub port: u16,
+}
+
+impl MockServer {
```

Comment:
Something to think of regarding the MockServer. Currently we are using a Mock server for lnurl testing. 
We are also dynamically replacingthe urls with a `maybe_replace_host_with_mock_test_host` function tha is only available for testing and registering the expected result on the mock server.
It also forces us to implement this service worker for wasm.
Since all lnurl requests are going through the function `get_parse_and_log_response` perhaps we should just create a testing version of it that returns the expected result and get rid of the mock server?
If we are able to do it it will remove a lot of specific WASM code and also the tests will use a lot of shared code.